src/map/ai/controllers/trust_controller.cpp

```diff
@@ -47,13 +47,13 @@ CTrustController::~CTrustController()
     }
     POwner->PAI->PathFind.reset();
     POwner->allegiance = ALLEGIANCE_PLAYER;
-    POwner->PMaster = nullptr;
-
+    POwner->status = STATUS_DISAPPEAR;
     m_LastTopEnmity = nullptr;
 }
 
 void CTrustController::Despawn()
 {
+    POwner->PMaster = nullptr;
```

Comment:
This seems to be a big source for instability around Trust lifetimes.
While the trust is despawning, disappearing and then eventually being deleted the reference to PMaster is alive right up until the Trust's destructor. There are many ticks happening in this window. Immediately setting it to nullptr (and a few paranoid checks later down the line) seem to have cleared up a lot of problems